src/lib.rs

```diff
@@ -191,22 +191,45 @@ impl Colorix {
         ui.add_space(20.);
         ui.vertical(|ui| {
             for (i, label) in LABELS.iter().enumerate() {
-                egui::ComboBox::from_label(*label)
-                    .selected_text(format!("{:?}", self.theme[i]))
-                    .show_ui(ui, |ui| {
-                        for j in 0..dropdown_colors.len() {
-                            if ui
-                                .selectable_value(
-                                    &mut self.theme[i],
-                                    dropdown_colors[j],
-                                    DROPDOWN_TEXT[j],
-                                )
-                                .clicked()
-                            {
-                                self.update_color(ctx, i);
-                            };
+                ui.horizontal(|ui| {
+                    if let Some(ColorPreset::Custom(rgb)) = self.theme.get_mut(i) {
+                        let re = ui.color_edit_button_srgb(rgb);
+                        if re.changed() {
+                            self.update_color(ctx, i);
                         }
-                    });
+                        let tokens = [
```

Comment:
This is kinda questionable, but I'm not sure how else to get the resulting tokens

Reply:
Quite. I'll think about it.

Reply:
> Quite. I'll think about it.

If you figure some API out, I can rebase my pull request on top of it

Reply:
You can remove this and instead of `token[i]` you should be able to use `self.theme[i].get_token()`

Reply:
Done